src/NovelRenderObject.cpp

```diff
@@ -11,19 +11,29 @@ NovelRenderObject::NovelRenderObject(NovelLayeringService* layeringService,
                                      const float screenScale,
                                      const GeoVector<float>& size,
                                      const NovelCommonArgs& args,
-                                     const GLuint programId) :
-    NovelObject(layeringService, screenScale, size, args), _programId(programId) {
+                                     const GLuint programId) : NovelObject(layeringService, screenScale, size, args),
+                                                               _programId(programId),
+                                                               _vertexArrayObject(Lazy<GLuint>([] {
+                                                                 GLuint tempVao;
+                                                                 glGenVertexArrays(1, &tempVao);
+                                                                 return tempVao;
+                                                               })),
+                                                               _buffer(Lazy<GLuint>([] {
+                                                                 GLuint tempBuffer;
+                                                                 glGenBuffers(1, &tempBuffer);
+                                                                 return tempBuffer;
+                                                               })) {
 }
 
 void NovelRenderObject::executeObjectBehaviour() {
-  if(!_bufferInitialised) {
-    configureObjectBuffers(true);
+  if (!_bufferInitialised) {
```

Comment:
Just to ensure I'm on the same page...

Now, if the buffer needs to be refreshed, you'll just reset `_bufferInitialised` to `false`, correct?

Reply:
Erm, good point actually. I had completely forgotten about this switch and will need to look further into this before this branch is merged.

Reply:
OK I've had a further look. This switch is there to ensure it is only initialised once via executing default object behaviour.

By "refresh" if you mean overwrite existing data in the buffers, that is the default behaviour of each object. If you mean to delete and generate new buffers, that is out of the scope of this PR with the changes introduced here. At the moment the only way to tear down buffers is to delete the object that manages it.

Perhaps you should open a ticket if you believe we should support having objects rebuild their buffers from scratch without also destroying the object in its entirety?

Reply:
Resolved in Discord.

Reply:
As I commented on Discord:
> ah, ok. I think I see.
Previously (current state, without the PR) refreshBuffers was used to decide if genBuffers should be called
now, you just rely on the buffer properly getting created via the lazy initializer
so, the parameter isn't needed
I was thinking it was also doing some other bits, but I'm just crazy
LGTM then